packages/coding-standard/src/Rules/CheckUnneededSymfonyStyleUsageRule.php

```diff
@@ -33,16 +46,27 @@ public function getNodeType(): string
      */
     public function processNode(Node $node, Scope $scope): array
     {
-        /** @var ObjectType $objectType */
-        $objectType = $scope->getType($node->var);
-        if (! is_a($objectType->getClassName(), SymfonyStyle::class, true)) {
-            return [];
+        /** @var MethodCall[] $methodCalls */
+        $methodCalls = $this->nodeFinder->findInstanceOf($node, MethodCall::class);
+        $foundAllowedMethod = [];
+        foreach ($methodCalls as $methodCall) {
+            /** @var Variable $variable */
+            $variable = $methodCall->var;
+            $objectType = $scope->getType($variable);
+            if ($objectType instanceof ObjectType && ! is_a($objectType->getClassName(), SymfonyStyle::class, true)) {
+                continue;
+            }
```

Comment:
This is actually false positive, because `$objectType` is always `MixedType`

Reply:
This one is a bit tricky, as the `Scope` belongs to `Class_`. So the `MethodCall` has no idea what type it is.

Reply:
thank you, it seems you handle at https://github.com/symplify/symplify/pull/2282 ?

Reply:
I tried but I didn't find a way :/

The problem is, only the MethodCall know types in their Scope. But we need to analyse class as a whole.

If you can come up with some solution, I'd appreciate it

Reply:
probably use static property/var to save the state for collection of methods call? I did try but still fail, probably need retry